server/asset/eth/rpcclient.go

```diff
@@ -48,24 +49,34 @@ type rpcclient struct {
 	tokens map[uint32]*tokener
 }
 
-func newRPCClient(net dex.Network, ipc string) *rpcclient {
+func newRPCClient(net dex.Network, endpoint string) *rpcclient {
 	return &rpcclient{
-		net:    net,
-		tokens: make(map[uint32]*tokener),
-		ipc:    ipc,
+		net:      net,
+		tokens:   make(map[uint32]*tokener),
+		endpoint: endpoint,
 	}
 }
 
 // connect connects to an ipc socket. It then wraps ethclient's client and
 // bundles commands in a form we can easily use.
 func (c *rpcclient) connect(ctx context.Context, log dex.Logger) error {
-	client, err := rpc.DialIPC(ctx, c.ipc)
-	if err != nil {
-		return fmt.Errorf("unable to dial rpc: %v", err)
+	var client *rpc.Client
+	var err error
+	if strings.HasSuffix(c.endpoint, ".ipc") {
+		client, err = rpc.DialIPC(ctx, c.endpoint)
+		if err != nil {
+			return fmt.Errorf("unable to dial ipc: %v", err)
+		}
+	} else {
+		log.Debugf("dialing endpoint: %s", c.endpoint)
+		client, err = rpc.Dial(c.endpoint)
```

Comment:
`DialContext(ctx, ...` please.  And will you also change the dial in multirpc while doing this?  Related: https://github.com/decred/dcrdex/issues/2045